pkg/repo/index.go

```diff
@@ -76,32 +74,139 @@ func (c ChartVersions) Less(a, b int) bool {
 	return i.LessThan(j)
 }
 
-// IndexFile represents the index file in a chart repository
-type IndexFile struct {
+// ChartRepositoryIndex represents the index file in a chart repository
+type ChartRepositoryIndex struct {
```

Comment:
This applies to several renames, but I'm just going to comment once for all of them.

The main issue with changing names like this is that it breaks API compatibility. "Officially", this would require us to bump Helm from version 2.x.y to 3.0.0, which we really don't want to do. This is required by the SemVer 2 spec that we committed to follow.

The official policy is that "No names will be changed in packages under `pkg`"

That said, we're talking about whether or not this particular package is isolated enough that nobody will really notice or care if we make these changes. So I guess I would ask you: Do you feel like these changes are _necessary_ or simply nice to have because they are clearer?